src/main/java/io/gingersnapproject/k8s/CacheRuleInformer.java

```diff
@@ -0,0 +1,232 @@
+         if (changedNewValues.size() > 0) {
+            // Update rule is unsupported and this should never happens. Assuption here is
+            // that the configMap
+            // is corrupted, so we raise an exception and no events are sent to the rule
+            // manager. Exception contains rule with old and new values
+            var changedOldValues = new HashSet<Map.Entry<String, String>>(olds);
+            changedOldValues.removeIf(
+                  arg0 -> !newMap.containsKey(arg0.getKey()) || arg0.getValue().equals(newMap.get(arg0.getKey())));
+            throw new UnsupportedOperationException("Rules cannot be updated: new values " + changedNewValues.toString()
+                  + ", old values " + changedOldValues.toString());
+         }
+      }
+
+      @Override
+      public void onDelete(ConfigMap obj, boolean deletedFinalStateUnknown) {
+         processConfigMapAndSend(obj.getData().entrySet(), (name, rule) -> {
+            log.debug("calling DynamicRuleManagement.removeRule({},{}", name, rule.toString());
+            drm.removeRule(name);
+         });
+      }
+   }
+}
+
+// Used only by CacheRuleInformer
+class EagerCacheRuleSpecAdapter implements Rule {
+   private final class ConnectorImplementation implements Connector {
+      private final String[] schemaTable;
+
+      private ConnectorImplementation(String[] schemaTable) {
+         this.schemaTable = schemaTable;
+      }
+
+      @Override
+      public String schema() {
+         return schemaTable[0];
+      }
+
+      @Override
+      public String table() {
+         return schemaTable[1];
+      }
+   }
+
+   final EagerCacheRuleSpec eagerRule;
+
+   public EagerCacheRuleSpecAdapter(EagerCacheRuleSpec eagerRule) {
+      this.eagerRule = eagerRule;
+   }
+
+   @Override
+   public Connector connector() {
+      String[] schemaTable = eagerRule.getTableName().split("\\.");
+      return switch (schemaTable.length) {
+         case 1 -> new Connector() {
+            @Override
+            public String schema() {
+               return "";
+            }
+
+            @Override
+            public String table() {
+               return schemaTable[0];
+            }
+         };
+         case 2 -> new ConnectorImplementation(schemaTable);
+         default ->
+            throw new IllegalArgumentException("Unsupported schema format (must be table or schema.table)");
+      };
+   }
+
+   @Override
+   public KeyFormat keyType() {
+      return eagerRule.getKey().getFormat();
+   }
+
+   @Override
+   public String plainSeparator() {
+      return eagerRule.getKey().getKeySeparator();
+   }
+
+   @Override
+   public List<String> keyColumns() {
+      return eagerRule.getKey().getKeyColumnsList();
+   }
+
+   @Override
+   public Optional<List<String>> valueColumns() {
+      return Optional.of(eagerRule.getValue().getValueColumnsList());
```

Comment:
We have to handle this slight differently. At the minute `getValueColumnsList` returns an empty list, not a null one if no value columns are defined in the k8s CR. Consequently, the following exception is thrown:

```java
2022-12-15 10:37:10,047 ERROR [io.fab.kub.cli.inf.imp.cac.SharedProcessor] (-1274506673-pool-9-thread-1) v1/namespaces/gingersnap-operator-system/configmaps failed invoking io.gingersnapproject.k8s.CacheRuleInformer$ConfigMapEventHandler@30a5aaac event handler: Number of arguments needs to be one or more: java.lang.IllegalArgumentException: Number of arguments needs to be one or more
	at io.gingersnapproject.cdc.translation.ColumnJsonTranslator.<init>(ColumnJsonTranslator.java:12)
	at io.gingersnapproject.cdc.cache.hotrod.HotRodService.backendForRule(HotRodService.java:89)
	at io.gingersnapproject.cdc.cache.hotrod.HotRodService_ClientProxy.backendForRule(Unknown Source)
	at io.gingersnapproject.cdc.EngineWrapper.start(EngineWrapper.java:94)
	at io.gingersnapproject.cdc.ManagedEngine$StartStopEngine.start(ManagedEngine.java:152)
	at io.gingersnapproject.cdc.ManagedEngine.addRule(ManagedEngine.java:110)
	at io.gingersnapproject.cdc.ManagedEngine_ClientProxy.addRule(Unknown Source)
	at io.gingersnapproject.k8s.CacheRuleInformer$ConfigMapEventHandler.lambda$onAdd$0(CacheRuleInformer.java:97)
	at io.gingersnapproject.k8s.CacheRuleInformer$ConfigMapEventHandler.processConfigMapAndSend(CacheRuleInformer.java:89)
	at io.gingersnapproject.k8s.CacheRuleInformer$ConfigMapEventHandler.onAdd(CacheRuleInformer.java:95)
	at io.gingersnapproject.k8s.CacheRuleInformer$ConfigMapEventHandler.onAdd(CacheRuleInformer.java:70)
	at io.fabric8.kubernetes.client.informers.impl.cache.ProcessorListener$AddNotification.handle(ProcessorListener.java:103)
	at io.fabric8.kubernetes.client.informers.impl.cache.ProcessorListener.add(ProcessorListener.java:50)
	at io.fabric8.kubernetes.client.informers.impl.cache.SharedProcessor.lambda$distribute$0(SharedProcessor.java:87)
	at io.fabric8.kubernetes.client.informers.impl.cache.SharedProcessor.lambda$distribute$1(SharedProcessor.java:110)
	at io.fabric8.kubernetes.client.utils.internal.SerialExecutor.lambda$execute$0(SerialExecutor.java:58)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)


```